aten/src/ATen/native/mps/operations/BitwiseBinaryOps.mm

```diff
@@ -0,0 +1,273 @@
+                                                     getMetalType(self),
+                                                     getMetalType(other),
+                                                     kernel_name);
+  uint32_t length = output.numel();
+  dispatch_sync(stream->queue(), ^(){
+    id<MTLCommandBuffer> buffer = stream->commandBuffer();
+    id<MTLComputeCommandEncoder> commandEncoder = [buffer computeCommandEncoder];
+
+    id<MTLBuffer> outBuf = __builtin_bit_cast(id<MTLBuffer>, output.storage().data());
+    id<MTLBuffer> selfBuf = __builtin_bit_cast(id<MTLBuffer>, self.storage().data());
+    id<MTLBuffer> otherBuf = __builtin_bit_cast(id<MTLBuffer>, other.storage().data());
+
+    [commandEncoder pushDebugGroup:[NSString stringWithFormat:@"Dispatch %s kernel", kernel_name.c_str()]];
+    [commandEncoder setComputePipelineState:cplState];
+    [commandEncoder setBytes:&length length:sizeof(length) atIndex:0];
+    [commandEncoder setBuffer:outBuf offset:output.storage_offset()*output.itemsize() atIndex:1];
+    [commandEncoder setBuffer:selfBuf offset:self.storage_offset()*self.itemsize()  atIndex:2];
+    [commandEncoder setBuffer:otherBuf offset:other.storage_offset()*other.itemsize() atIndex:3];
+    dispatch1DJob(commandEncoder, cplState, length);
+    [commandEncoder endEncoding];
+    stream->commit(true);
+  });
+}
+
+void handle_tensor_scalar_binary_op(const at::Tensor& self, const at::Scalar& other, at::Tensor& output, const std::string& kernel_name) {
+  using namespace at::mps;
+  MPSStream* stream = getCurrentMPSStream();
+  id<MTLComputePipelineState> cplState = getCPLState(MPSDevice::getInstance()->device(),
+                                                     getMetalType(output),
+                                                     getMetalType(self),
+                                                     getMetalType(other),
+                                                     kernel_name);
+  uint64_t sval = other.to<int64_t>();
+  uint32_t length = output.numel();
+  dispatch_sync(stream->queue(), ^(){
+    id<MTLCommandBuffer> buffer = stream->commandBuffer();
+    id<MTLComputeCommandEncoder> commandEncoder = [buffer computeCommandEncoder];
+
+    id<MTLBuffer> outBuf = __builtin_bit_cast(id<MTLBuffer>, output.storage().data());
+    id<MTLBuffer> selfBuf = __builtin_bit_cast(id<MTLBuffer>, self.storage().data());
+
+    [commandEncoder pushDebugGroup:[NSString stringWithFormat:@"Dispatch %s kernel", kernel_name.c_str()]];
+    [commandEncoder setComputePipelineState:cplState];
+    [commandEncoder setBytes:&length length:sizeof(length) atIndex:0];
+    [commandEncoder setBuffer:outBuf offset:output.storage_offset()*output.itemsize() atIndex:1];
+    [commandEncoder setBuffer:selfBuf offset:self.storage_offset()*self.itemsize()  atIndex:2];
+    [commandEncoder setBytes:&sval length:sizeof(sval) atIndex:3];
+    [commandEncoder dispatchThreadgroups:MTLSizeMake((length + 511) / 512, 1, 1)
```

Comment:
Should be dispatch1DJob as well?

Reply:
https://github.com/pytorch/pytorch/pull/82982